runtime/lua/vim/lsp/util.lua

```diff
@@ -1417,6 +1417,11 @@ function M._make_floating_popup_size(contents, opts)
   -- make sure borders are always inside the screen
   width = math.min(width, screen_width - border_width)
 
+  -- Make sure that the width is large enough to fit the title.
+  if opts.title then
+    width = math.max(width, vim.fn.strdisplaywidth(opts.title))
```

Comment:
This causes an error when using tuple for the title

```lua
vim.diagnostic.open_float({
  title = { { 'Hello', 'DiagnosticInfo' } },
})
```

```
E5108: Error executing lua Vim:E730: Using a List as a String
stack traceback:
        [C]: in function 'strdisplaywidth'
        /home/fitrh/src/neovim/runtime/lua/vim/lsp/util.lua:1415: in function '_make_floating_popup_size'
        /home/fitrh/src/neovim/runtime/lua/vim/lsp/util.lua:1599: in function 'open_floating_preview'
        /home/fitrh/src/neovim/runtime/lua/vim/diagnostic.lua:2334: in function 'open_float'
        [string ":lua"]:1: in main chunk
```

Also, it seems that the docs for `:h vim.lsp.util.open_floating_preview.Opts` missed annotating the `title` as `string|[string,string]`